forage-android/src/main/java/com/joinforage/forage/android/network/data/CheckBalanceRepository.kt

```diff
@@ -74,27 +75,24 @@ internal class CheckBalanceRepository(
         var attempt = 1
 
         while (true) {
-            logger.debug("Polling check balance message status. Attempt: $attempt.")
+            logger.i("Polling for balance check response for Payment Method $paymentMethodRef")
 
             when (val response = messageStatusService.getStatus(contentId)) {
                 is ForageApiResponse.Success -> {
                     val balanceMessage = Message.ModelMapper.from(response.data)
 
                     if (balanceMessage.status == "completed") {
-                        logger.debug("Status is completed.")
                         if (balanceMessage.failed) {
-                            logger.debug("Failed is true.")
                             val error = balanceMessage.errors[0]
+                            logger.e("Received response ${error.statusCode} for balance check of Payment Method $paymentMethodRef with message: ${error.message}")
```

Comment:
include `contentId` in log? Also, worth including `paymentMethodRef` in log tag / Context for easier DataDog search?